src/components/transactions/Emode/EmodeModalContent.tsx

```diff
@@ -40,78 +43,43 @@ export interface EmodeModalContentProps {
   mode: EmodeModalType;
 }
 
+function getInitialEmode(
+  mode: EmodeModalType,
+  eModes: AppDataContextType['eModes'],
+  currentEmode: number
+) {
+  const eModesNumber = Object.keys(eModes).length;
+  if (mode === EmodeModalType.ENABLE) {
+    if (eModesNumber > 2) return undefined;
+    return eModes[1];
+  }
+  if (mode === EmodeModalType.SWITCH) {
+    if (eModesNumber > 3) return undefined;
+    if (currentEmode === 1) return eModes[2];
+    return eModes[1];
+  }
+  return eModes[0];
+}
+
 export const EmodeModalContent = ({ mode }: EmodeModalContentProps) => {
   const {
     user,
     reserves,
+    eModes,
     marketReferenceCurrencyDecimals,
     marketReferencePriceInUsd,
     userReserves,
   } = useAppDataContext();
-  const { currentChainId, currentNetworkConfig } = useProtocolDataContext();
+  const { currentChainId } = useProtocolDataContext();
   const { chainId: connectedChainId } = useWeb3Context();
   const currentTimestamp = useCurrentTimestamp(1);
   const { gasLimit, mainTxState: emodeTxState, txError } = useModalContext();
 
-  const [selectedEmode, setSelectedEmode] = useState<EmodeCategory | undefined>(undefined);
-  const [emodeCategories, setEmodeCategories] = useState<Record<number, EmodeCategory>>({});
-
+  const [selectedEmode, setSelectedEmode] = useState<EmodeCategory | undefined>(
+    getInitialEmode(mode, eModes, user.userEmodeCategoryId)
+  );
   const networkConfig = getNetworkConfig(currentChainId);
 
-  // Create object of available emodes
-  useEffect(() => {
```

Comment:
this code was quite literally hell to read and the useEffect was kinda unnecessary